xxh3.h

```diff
@@ -1509,18 +1512,101 @@ XXH_FORCE_INLINE void XXH3_initCustomSecret(xxh_u8* XXH_RESTRICT customSecret, x
      */
     XXH_ASSERT(kSecretPtr == kSecret);
 
-    for (i=0; i < nbRounds; i++) {
+#if (XXH_VECTOR == XXH_AVX512)
+    XXH_STATIC_ASSERT((XXH_SECRET_DEFAULT_SIZE & 63) == 0);
+    XXH_STATIC_ASSERT(XXH_SEC_ALIGN <= 64);
+    (void)kSecretPtr;
+    (void)(&XXH_writeLE64);
+    {   int const nbRounds = XXH_SECRET_DEFAULT_SIZE / sizeof(__m512i);
+        __m512i const seed = _mm512_mask_set1_epi64(_mm512_set1_epi64((xxh_i64)seed64), 0xAA, -(xxh_i64)seed64);
+
+        XXH_ALIGN(64) const __m512i* const src  = (const __m512i*) kSecret;
+        XXH_ALIGN(64)       __m512i* const dest = (      __m512i*) customSecret;
+        for (i=0; i < nbRounds; ++i) {
+            // GCC has a bug, _mm512_stream_load_si512 accepts 'void*', not 'void const*',
+            // this will warn "discards ‘const’ qualifier".
+            union {
+                XXH_ALIGN(64) const __m512i* const cp;
+                XXH_ALIGN(64) void* const p;
+            } const remote_const_void = { .cp = src + i };
+            dest[i] = _mm512_add_epi64(_mm512_stream_load_si512(remote_const_void.p), seed);
+        }
+    }
+#elif (XXH_VECTOR == XXH_AVX2)
+    XXH_STATIC_ASSERT((XXH_SECRET_DEFAULT_SIZE & 31) == 0);
+    XXH_STATIC_ASSERT((XXH_SECRET_DEFAULT_SIZE / sizeof(__m256i)) == 6);
+    XXH_STATIC_ASSERT(XXH_SEC_ALIGN <= 64);
+    (void)kSecretPtr;
+    (void)(&XXH_writeLE64);
+    (void)i;
+    XXH_PREFETCH(customSecret);
```

Comment:
minor : what's the benefit of this prefetch ?
`customSecret` is merely written into, and not read from.

Reply:
`customSecret` will be read later when perform the hashing, so when writing the secret, we should not use non-temporal instructions, then the `customSecret` will always be mapped into cache even for write-only access. The prefetch here makes the some of the following assignment a little faster.

Reply:
I note that the `prefetch` instruction is used on the `AVX2` path, but not the other ones (neither `SS2`, `AVX512` nor `scalar`). Is there a reason for that difference ?

Reply:
I have tried to enable prefetching with different lengths for `SSE2` and `AVX512`, and do not get reproducible improvements, while `AVX2` did.

For `scalar` branch, it produces exactly the same code after this patch as before, so do not enable prefetch.

Reply:
BTW. For AVX512 if you use non-temporal loads (NTA, aka stream loads), it is probably also useful to use non-temporal prefetch instruction. This way you actually protect the cache against evictions, otherwise you will nullify benefits of non-temporarl loads by using wrong prefetch. Just my thoughts. Performance advantages of nta prefetch and loads are very hard to measure, because it impacts both the code doing loads (it could be a bit faster, or maybe use less power), but even if it has the same speed, it might impact positively other code, processes or kernel (by not trashing L1 / L2 caches), which is harder to measure.